api/src/modules/tasks/tasks.controller.ts

```diff
@@ -1,4 +1,104 @@
-import { Controller } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiForbiddenResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import {
+  FetchSpecification,
+  ProcessFetchSpecification,
+} from 'nestjs-base-service';
+import { PaginationMeta } from 'utils/app-base.service';
+import { Task, taskResource } from 'modules/tasks/task.entity';
+import { TasksService } from 'modules/tasks/tasks.service';
+import { UpdateTaskWithControllerDto } from 'modules/tasks/dto/update-task-with-controller.dto';
+import { CreateTaskDto } from 'modules/tasks/dto/create-task.dto';
 
-@Controller('tasks')
-export class TasksController {}
+@Controller('/api/v1/tasks')
+@ApiTags(taskResource.className)
+export class TasksController {
+  constructor(protected readonly taskService: TasksService) {}
+
+  @ApiOperation({ description: 'Find all tasks' })
+  @ApiOkResponse({ type: Task })
+  @ApiBadRequestResponse()
+  @ApiUnauthorizedResponse()
+  @ApiForbiddenResponse()
+  @Get()
+  async findAll(
+    // To implement once Auth is merged:
+    //@Req() request: Request,
+    @ProcessFetchSpecification({
+      allowedFilters: taskResource.columnsAllowedAsFilter,
+    })
+    fetchSpecification: FetchSpecification,
+  ): Promise<Task> {
+    // To implement once Auth is merged:
```

Comment:
not really sure what you are trying to do here.

Reply:
I believe the intention here is to add the ability to return tasks created by a given user, with the ID coming in the request once auth is done.
As mentioned before, the proper approach to this would be based on a role, but I think we can work with this. 
Right now the API returns a task once a import job is pushed to the queue and we also provide a way to get that task by it's ID, but I think we should keep this as well

Reply:
yes, I added a GetUser decorator which is supposed to get user object from the request (after Auth is done), then the idea is to add user id to the 'filters' property of fetchSpecification object, which is passed to findAllPaginated() method of nestjs-base-service, so that it returns all Tasks created By the authenticated User

Reply:
ok, but I don't agree with returning just the tasks for a given user. At least for our only Task type right now, it is a process that affects all users, so I don't really see a motivation to hide other user's tasks. Maybe that's something we will need to add in the future, but for now I don't think it makes sense.

Reply:
okay, removed